test/functional/fixtures/api/es-next/iframe-switching/testcafe-fixtures/iframe-switching-test.js

```diff
@@ -52,6 +52,21 @@ test('Click on element in a nested iframe', async t => {
     expect(iframeBtnClickCount).eql(1);
 });
 
+test.page`http://localhost:3000/fixtures/api/es-next/iframe-switching/pages/shadow.html`
+('Click on an element in a shadow iframe and return to the main window', async t => {
+    await t
```

Comment:
Add a test for the case with nested shadow roots with iframes:
- element
    - shadowRoot
        - iframe 
            - element
                - shadowRoot
                     - iframe